connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java

```diff
@@ -77,6 +92,7 @@ public synchronized void start() {
     @Override
     public synchronized void stop() {
         log.info("Herder stopping");
+        requestExecutorService.shutdown();
```

Comment:
aren't we missing a corresponding `awaitTermination`?

Reply:
I've added an `awaitTermination` of 30 secs